core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/DatabaseUtil.java

```diff
@@ -598,13 +636,42 @@ private static void closeStatement(PreparedStatement preparedStatement) {
         if (preparedStatement != null) {
             try {
                 preparedStatement.close();
+            } catch (SQLRecoverableException ex) {
+                handleSQLRecoverableException(ex);
             } catch (SQLException e) {
                 log.error("Database error. Could not close statement. Continuing with others. - " + e.getMessage(), e);
             }
         }
 
     }
 
+    private static void handleSQLRecoverableException(SQLRecoverableException ex){
+		Connection dbConnection;
+    	try {
+	    	if(dataSource != null){
+	    		dbConnection = getDBConnection(dataSource);
+				handleSQLRecoverableException(dbConnection, ex);
+	    	}
+    	} catch (SQLException e) {
+            log.error("Error occurred during SQLRecoverableException handling.", e);
+    	}
+    }
+    
+    private static void handleSQLRecoverableException(Connection dbConnection, SQLRecoverableException ex) {
```

Comment:
Method name handleSQLRecoverableException() does not imply that it is trying to recover a close() operation. On the other hand, would it require to retry a database connection close operation? Would you mind explaining your intention here?